ip_country/src/country_finder.rs

```diff
@@ -0,0 +1,301 @@
+    }
+
+    #[test]
+    fn does_not_find_ipv4_address_in_zz_block() {
+        let timestart = SystemTime::now();
+        let result = CountryCodeFinder::find_country(
+            &COUNTRY_CODE_FINDER_TEST,
+            IpAddr::from_str("0.0.5.0").unwrap(),
+        );
+        let timeend = SystemTime::now();
+
+        assert_eq!(result, None);
+        let duration = timeend.duration_since(timestart).unwrap().as_secs();
+        println!("duration: {}", duration);
+        assert!(duration <= 5);
+    }
+
+    #[test]
+    fn finds_ipv6_address_in_fourth_block() {
+        let result = CountryCodeFinder::find_country(
+            &COUNTRY_CODE_FINDER_TEST,
+            IpAddr::from_str("1:0:5:0:0:0:0:0").unwrap(),
+        );
+
+        assert_eq!(result, Some(Country::try_from("AU").unwrap()))
+    }
+
+    #[test]
+    fn does_not_find_ipv6_address_in_zz_block() {
+        let result = CountryCodeFinder::find_country(
+            &COUNTRY_CODE_FINDER_TEST,
+            IpAddr::from_str("0:0:5:0:0:0:0:0").unwrap(),
+        );
+
+        assert_eq!(result, None)
+    }
+
+    #[test]
+    fn real_test_ipv4_with_google() {
+        let result = CountryCodeFinder::find_country(
+            &COUNTRY_CODE_FINDER,
+            IpAddr::from_str("142.250.191.132").unwrap(), // dig www.google.com A
+        )
+        .unwrap();
+
+        assert_eq!(result.name, "United States".to_string());
+    }
+
+    #[test]
+    fn real_test_ipv4_with_cz_isp() {
+        let result = CountryCodeFinder::find_country(
+            &COUNTRY_CODE_FINDER,
+            IpAddr::from_str("77.75.77.222").unwrap(), // dig www.seznam.cz A
+        )
+        .unwrap();
+
+        assert_eq!(result.free_world, true);
+        assert_eq!(result.iso3166, "CZ".to_string());
+        assert_eq!(result.name, "Czechia".to_string());
+    }
+
+    #[test]
+    fn real_test_ipv4_with_sk_isp() {
+        let timestart = SystemTime::now();
+        let result = CountryCodeFinder::find_country(
+            &COUNTRY_CODE_FINDER,
+            IpAddr::from_str("213.81.185.100").unwrap(), // dig www.zoznam.sk A
+        )
+        .unwrap();
+        let timeend = SystemTime::now();
+
+        assert_eq!(result.free_world, true);
+        assert_eq!(result.iso3166, "SK".to_string());
+        assert_eq!(result.name, "Slovakia".to_string());
+        let duration = timeend.duration_since(timestart).unwrap().as_secs();
+        println!("duration: {}", duration);
+        assert!(duration <= 5);
+    }
+
+    #[test]
+    fn real_test_ipv6_with_google() {
+        let timestart = SystemTime::now();
+        let result = CountryCodeFinder::find_country(
+            &COUNTRY_CODE_FINDER,
+            IpAddr::from_str("2607:f8b0:4009:814::2004").unwrap(), // dig www.google.com AAAA
+        )
+        .unwrap();
+        let timeend = SystemTime::now();
+
+        assert_eq!(result.name, "United States".to_string());
```

Comment:
You could possibly also add the two more params to be asserted on the CountryCode.